clients/include/utility.hpp

```diff
@@ -291,6 +291,43 @@ class hipsolver_local_syevj_info
     }
 };
 
+/* ============================================================================================
+ */
+/*! \brief  local params which is automatically created and destroyed  */
+class hipsolver_local_params
+{
+    hipsolverDnParams_t m_info;
+
+public:
+    hipsolver_local_params()
+    {
+        if(hipsolverDnCreateParams(&m_info) != HIPSOLVER_STATUS_SUCCESS)
+            throw std::runtime_error("ERROR: Could not create hipsolverDnParams_t");
+    }
+    ~hipsolver_local_params()
+    {
+        hipsolverDnDestroyParams(m_info);
+    }
+
+    hipsolver_local_params(const hipsolver_local_params&) = delete;
+
+    hipsolver_local_params(hipsolver_local_params&&) = delete;
+
+    hipsolver_local_params& operator=(const hipsolver_local_params&) = delete;
+
+    hipsolver_local_params& operator=(hipsolver_local_params&&) = delete;
```

Comment:
I'm not terribly familiar with how move operations work, but I think the idea is to ensure that the class deconstructor can only ever be called once, since that will deallocate the wrapped Params object.

Reply:
The move operations could be implemented, but it's more complicated than just deleting them. Since we don't actually use the move operations, we might as well just keep it simple.